pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -727,15 +729,25 @@ func (l *LibvirtDomainManager) asyncMigrate(vmi *v1.VirtualMachineInstance, opti
 		monitor := newMigrationMonitor(vmi, l, options, migrationErrorChan)
 		go monitor.startMonitor()
 
-		params, err := generateMigrationParams(dom, vmi, options)
-		if err != nil {
-			log.Log.Object(vmi).Reason(err).Error("Live migration failed while generating migration parameters")
-			migrationErrorChan <- err
-			return
+		// anything that modifies the domain needs to be performed with the domainModifyLock held
+		// The domain params and unHotplug need to be performed in a critical section together.
+		critSection := func() error {
```

Comment:
I would prefer if the whole PR is rebased on PR#5230 from start, as in practice changes there have been dropped/changed  when moving the code.
It may be ok to change the code, but it needs to be transparent. At the moment, the move of the code is both moving and changing it, unfortunately it is not easy to examine it.

Reply:
it's squashed